### PR TITLE
Generate a using statement when starting a new GlobalSuppressions file

### PR DESCRIPTION
**Pull request by jnm2**
This is always the first thing I do to clean up after it.

### CODE REVIEW

**Comment by JoeRobich** [comment]
Thanks @jnm2!

**Comment by jnm2** [comment]
Love the quick response, thank you!

File: src/Features/CSharp/Portable/CodeFixes/Suppression/CSharpSuppressionCodeFixProvider.cs [hunk 1]
```diff
@@ -87,16 +88,46 @@ protected override bool IsEndOfLine(SyntaxTrivia trivia)
...
+
+            var fullAttributeName = (QualifiedNameSyntax)SyntaxFactory.ParseName(SuppressMessageAttributeName);
+            var usingDirective = SyntaxFactory.UsingDirective(fullAttributeName.Left);
+
+            var hasUsingDirective = addImportsService.HasExistingImport(compilation, compilationRoot, contextLocation: null, usingDirective);
+            if (!hasUsingDirective && isFirst)
+            {
+                compilationRoot = Format(compilationRoot.AddUsings(usingDirective));
+                hasUsingDirective = true;
+            }
```
Comment (sharwell) [inline src/Features/CSharp/Portable/CodeFixes/Suppression/CSharpSuppressionCodeFixProvider.cs]: ❓ Can we not just use `Simplifier.AddImportsAnnotation` on the attribute?

Comment (mavasani) [reply src/Features/CSharp/Portable/CodeFixes/Suppression/CSharpSuppressionCodeFixProvider.cs]: Agreed. Explicitly invoking the formatter here seems strange.

Comment (jnm2) [reply src/Features/CSharp/Portable/CodeFixes/Suppression/CSharpSuppressionCodeFixProvider.cs]: @mavasani In 2015 you disabled formatting and simplifying in the fix-all and started calling the formatted here. See https://github.com/dotnet/roslyn/pull/6232/files#diff-6d2fa3555fa9d81aa6f841a1c444ab2dR49

Comment (jnm2) [reply src/Features/CSharp/Portable/CodeFixes/Suppression/CSharpSuppressionCodeFixProvider.cs]: @sharwell I tried that first and when it didn't work I discovered the perf note linked above.

**Comment by CyrusNajmabadi** [comment]
Working on a fix currently.

File: src/Features/CSharp/Portable/CodeFixes/Suppression/CSharpSuppressionCodeFixProvider.cs (hunk 1 above)
Comment (mavasani) [reply src/Features/CSharp/Portable/CodeFixes/Suppression/CSharpSuppressionCodeFixProvider.cs]: @jnm2 Oh thanks for finding that out :-). I now recall I originally did not have the Format invocation, but invoking a FixAll caused massive performance overhead and had to fallback to invoking Format explicitly.

Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/CodeFixes/Suppression/CSharpSuppressionCodeFixProvider.cs]: Fix is here: https://github.com/dotnet/roslyn/pull/41226

Comment (sharwell) [reply src/Features/CSharp/Portable/CodeFixes/Suppression/CSharpSuppressionCodeFixProvider.cs]: @CyrusNajmabadi That link goes to this PR lol

Comment (CyrusNajmabadi) [reply src/Features/CSharp/Portable/CodeFixes/Suppression/CSharpSuppressionCodeFixProvider.cs]: Oops.  https://github.com/dotnet/roslyn/pull/41239

**Pull request merged**